src/main/java/de/tum/in/www1/artemis/web/rest/PersonalAccessTokenResource.java

```diff
@@ -0,0 +1,79 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.security.UserNotActivatedException;
+import de.tum.in.www1.artemis.security.jwt.TokenProvider;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+/**
+ * Controller to request personal access tokens.
+ */
+@RestController
+@RequestMapping("/api")
+@Profile("pat")
+public class PersonalAccessTokenResource {
+
+    private static final String ENTITY_NAME = "personalAccessTokenResource";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final TokenProvider tokenProvider;
+
+    private final UserRepository userRepository;
+
+    @Value("${artemis.personal-access-token.max-lifetime-in-days:30}")
+    private long personalAccessTokenMaxLifetimeDays;
+
+    public PersonalAccessTokenResource(TokenProvider tokenProvider, UserRepository userRepository) {
+        this.tokenProvider = tokenProvider;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * Generates a personal access token (JWT token) for the current user.
+     *
+     * @param lifetimeDays the lifetime for the JWT token in days
+     * @return a JWT Token
+     */
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping("/personal-access-token")
+    public ResponseEntity<UserJWTController.JWTToken> getPersonalAccessToken(@RequestBody Long lifetimeDays) {
```

Comment:
```suggestion
    public ResponseEntity<UserJWTController.JWTToken> getPersonalAccessToken(@RequestBody Long requestedLifetimeDays) {
```
Makes it more clear what this is.